src/components/projectSelect/index.tsx

```diff
@@ -4,35 +4,63 @@ import Cookies from "js-cookie";
 import { useHistory } from "react-router-dom";
 import SearchableDropdown from "components/SearchableDropdown";
 import { CURRENT_PROJECT } from "constants/cookies";
-import { getCommitsRoute } from "constants/routes";
 import {
   GetProjectsQuery,
   GetProjectsQueryVariables,
+  GetViewableProjectRefsQuery,
+  GetViewableProjectRefsQueryVariables,
 } from "gql/generated/types";
-import { GET_PROJECTS } from "gql/queries";
+import { GET_PROJECTS, GET_VIEWABLE_PROJECTS } from "gql/queries";
 import { ProjectOptionGroup } from "./ProjectOptionGroup";
 
+type project = {
+  displayName: string;
+  identifier: string;
+  repoRefId: string;
+  isFavorite: boolean;
+};
 interface ProjectSelectProps {
   selectedProjectIdentifier: string;
+  isProjectSettingsPage?: boolean;
+  getRoute: (projectIdentifier: string) => string;
 }
 export const ProjectSelect: React.FC<ProjectSelectProps> = ({
   selectedProjectIdentifier,
+  isProjectSettingsPage = false,
+  getRoute,
 }) => {
-  const { data, loading } = useQuery<
+  const { data: projectsData, loading: projectsLoading } = useQuery<
     GetProjectsQuery,
     GetProjectsQueryVariables
-  >(GET_PROJECTS);
+  >(GET_PROJECTS, {
+    skip: isProjectSettingsPage,
+  });
+
+  const {
+    data: viewableProjectsData,
+    loading: viewableProjectsLoading,
+  } = useQuery<
+    GetViewableProjectRefsQuery,
+    GetViewableProjectRefsQueryVariables
+  >(GET_VIEWABLE_PROJECTS, {
+    skip: !isProjectSettingsPage,
+  });
+
+  const projects = getProjects(projectsData, viewableProjectsData);
+  const loading = viewableProjectsLoading || projectsLoading;
+
   const history = useHistory();
-  const { projects } = data || { projects: [] };
 
-  const favoriteProjects = projects?.flatMap((g) =>
+  const favoriteProjects = projects.flatMap((g) =>
     g.projects.filter((p) => p.isFavorite)
   );
 
-  const allProjects = [
-    { name: "Favorites", projects: favoriteProjects },
-    ...projects,
-  ];
+  const favorites =
+    favoriteProjects.length > 0
+      ? { name: "Favorites", projects: favoriteProjects }
+      : {};
+
+  const allProjects = [favorites, ...projects];
```

Comment:
```suggestion
  const favorites = { name: "Favorites", projects: favoriteProjects }

  const allProjects = favoriteProjects.length > 0 ? [favorites, ...projects] : projects;
```

Reply:
This avoids adding an empty object to allProjects.

Reply:
I question if this is even necessary? I would imagine we would always want to show people their favorite projects even on the project settings page.

Reply:
I'm not sure which part you are questioning is necessary. We do show people their favorite projects even on the project settings page.

Reply:
your suggestion results in type errors later on.

Reply:
nvm, it doesn't

Reply:
I'm questioning the code changes around the favoriteProjects logic in that case.